rfd/9999-aws-api-integration-using-oidc.md

```diff
@@ -0,0 +1,554 @@
+
+When configuring the provider, we'll need an AWS Role which Teleport uses to issue API Calls.
+
+To store the configuration above, we'll create a new resource Kind: `Integration`.
+It'll leverage the `subkind` prop to distinguish future integrations.
+
+### High Level Flow
+Simplified flow of interactions between User, Teleport and AWS:
+
+```
+                                                               ┌─────────┐
+                                                               │         │
+                ┌─────────────────────────────────────────────►│  AWS    │
+               2│                                              │         │
+                │                                              └──┬───▲──┘
+User────────────┤                                                 │   │
+                │       ┌───────────────────────┐                 │   │
+               1│       │       Teleport        │                 │   │
+                │       │                       │                 │   │
+               5│    ┌──┼──────┐       ┌────────┼──┐              │   │
+                └────►  │ Web  │       │ OIDC   │  │       3      │   │
+                     └──┼──┬───┘       │ IdP    │  │◄─────────────┘   │
+                        │  │           └┬───────┼──┘                  │
+                        │  │            │       │                     │
+                        │  │           4│       │                     │
+                        │  │            │       │                     │
+                        │  │  ┌─────────▼┐      │                     │
+                        │  │  │ CA KeySet│      │                     │
+                        │  │  │ +OIDC CA │      │                     │
+                        │  │  └─────────▲┘      │                     │
+                        │  │6           │7      │                     │
+                        │  │     ┌──────┴────┐  │                     │
+                        │  │     │ Discover  │  │           8         │
+                        │  └────►│ Service   ├──┼─────────────────────┘
+                        │        │ [ RDS ]   │  │
+                        │        └───────────┘  │
+                        │                       │
+                        └───────────────────────┘
+```
+
+#### User's Point of View
+When the user adds a new resource, using the Discover Wizard, they'll be asked where the resource lives.
+Choosing an RDS DB, the user is presented with the following flow:
+
+1. User is prompted to setup an OIDC Identity Provider pointing to its Teleport instance's url.
+2. User opens AWS, selects IAM and then, under `Access Management`, selects `Identity providers`.
+3. Configures the Teleport instance as an OIDC IdP:
+   1. Clicks `Add provider` of type `OpenID Connect`.
+   2. Fills in the Provider URL as provided by the Discover Wizard.
+   3. Fills in the Audience as provided by the Discover Wizard `sts.amazonaws.com`.
```

Comment:
Does the audience have to be `sts.amazonaws.com`? We could skip this step if we had a fixed audience as part of the OIDC IdP in Teleport.

Reply:
Whilst AWS doesn't enforce it, and I'd argue they should, `sts.amazonaws.com` definitely makes the most semantic sense for the value here as that's the host of the AWS service that allows the OIDC ID token to be swapped for AWS credentials. I imagine they don't enforce it universally as not too long ago some CI/CD providers did not allow you to customise the audience of issued tokens to workloads.